fixes that you can do what you want with (#3)
FSG-Cat/matrix-spec-proposals: proposals/XXXX-incremental-heartbeat-assisted-presence.md

All homeserver implementations recognise presence as a very inefficient feature, and also a feature that does not scale.
In addition, the current presence system is fragile: there have been presence bugs in sliding sync proxy, and presence flooding
can be accidentally triggered by clients.

### High level overview of how to fix this.

The High level overview goes into a higher level overview of this MSC and goes into background.
The overview is provided for the purpose of making the MSC easier to write, and to make the proposal easier to understand.
As the high level overview explains the system without getting too concerned with the minute details found in the proposal section.

This MSC proposes fixing presence by taking a page out of the routing protocol world. The precursor to
this MSC in the form of the gist from earlier exists due to my studies having us learn the ins and outs of
the EIGRP and OSPF routing protocols. It turns out that these routing protocols have a similar problem to us.
They need to distribute Alive or Dead status to all the nodes so your routing tables are up to date.

> And what is presence if not Alive or Dead status information? Now OSPF takes the problematic-but-works-at-limited

Reviewer: ```suggestion
And what is presence if not Alive or Dead status information? Now OSPF takes the problematic, although working at a limited scale,
```

> enough-scale approach of sending a complete copy of state every time anything changes. Its a known flaw but it works.

Reviewer: ```suggestion
approach of sending a complete copy of state every time anything changes. Its a known flaw but it works.
```

Now EIGRP uses incremental updates and that's exactly what i propose we copy. This MSC therefore proposes
that we change presence to be a incremental system that only communicates state changes and move expiry into a

> separate subsystem of presence, we have called the heartbeat system.

Reviewer: I am not completely sure im happy with this but my inclanation is F it. Looks good enough to me.

To facilitate incremental presence we only need to keep track of whether a given party is alive or dead. For example,
if matrix.org is upgrading their database over a duration of six hours, we wouldn't want matrix.org's users to be
marked as online for the entire maintenance duration.
The proposal section goes into detail around how this alive tracking works but in the high level overview its
also important to include that servers can tell other servers that they are going offline/disabling presence.

> So how does changes-only and knowing-alive-or-dead status for homeservers help? Well simple, if one takes a look at

Reviewer: ```suggestion
So how does changes-only and if a homeserver is offline or online help? Well simple, if one takes a look at
```
This is a potential change but i am not completely sure about it. Could use a second pass tbh.

All i know is that the dashes have to go in the dead or alive section. But changes-only that im fine with. It doesnt sound wrong to the point of me being "sick"/s
